pkg/pgmodel/ingestor/dispatcher.go

```diff
@@ -228,15 +226,27 @@ func (p *pgxDispatcher) InsertData(rows map[string][]model.Samples) (uint64, err
 			close(errChan)
 			if err != nil {
 				log.Error("msg", fmt.Sprintf("error on async send, dropping %d datapoints", numRows), "err", err)
-			} else if p.insertedDatapoints != nil {
-				atomic.AddInt64(p.insertedDatapoints, int64(numRows))
 			}
+			postIngestTasks(maxt, numRows)
 		}()
 	}
 
 	return numRows, err
 }
 
+// postIngestTasks performs a set of tasks that are due after ingesting series data.
+func postIngestTasks(ts, numRows int64) {
```

Comment:
may be better in ingestor/metrics.go

Reply:
Why will post ingest task go inside a metrics.go file?